File: 프로그래머스/unrated/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.java
class Solution {
    public int solution(String t, String p) {
        int answer = 0;
        int plen = p.length();
        int tlen = t.length();
        double pow = Math.pow(10,plen);
        long pi = Long.parseLong(p);
        for(int i=0; i<=tlen-plen; i++){
            if(Long.parseLong(t.substring(i, i+plen)) <= pi)
                answer++;
        }
        return answer;
    }
}

File: 프로그래머스/unrated/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.js
function solution(t, p) {
    var answer = 0;
    let plen = p.length;
    p = Number(p);
    for(var i=0; i<=t.length-plen; i++){
        let tsub = Number(t.substr(i, plen));
        if(p >= tsub) answer++;
    }
    return answer;
}